sql/src/main/java/org/apache/druid/sql/calcite/schema/SystemSchema.java

```diff
@@ -412,21 +418,16 @@ public TableType getJdbcTableType()
       final List<ImmutableDruidServer> druidServers = serverView.getDruidServers();
       final AuthenticationResult authenticationResult =
           (AuthenticationResult) root.get(PlannerContext.DATA_CTX_AUTHENTICATION_RESULT);
+      final Access access = AuthorizationUtils.authorizeAllResourceActions(
+          authenticationResult,
+          Collections.singletonList(new ResourceAction(new Resource("STATE", ResourceType.STATE), Action.READ)),
+          authorizerMapper
+      );
+      if (!access.isAllowed()) {
+        return Linq4j.asEnumerable(ImmutableList.of());
```

Comment:
Since this is an "all or nothing" check, I think it'd be better to throw a ForbiddenException here and let the user know they have insufficient privileges to see the servers

Reply:
okay will throw exception.